packages/legend-graph/src/models/protocols/pure/v1/engine/V1_Engine.ts

```diff
@@ -471,12 +472,16 @@ export class V1_Engine {
     generationMode: GenerationMode,
     model: V1_PureModelContextData,
   ): Promise<V1_GenerationOutput[]> {
+    const grammar = this.pureModelContextDataToPureCode(model);
+    const pureModelContextText = new V1_PureModelContextText();
+    pureModelContextText.serializer = model.serializer;
+    pureModelContextText.code = await grammar;
```

Comment:
let's be a little bit more idiomatic here :) `await grammar` is odd; we'd better do `await this.pureModelContextDataToPureCode(model);`